doc/index.md

```diff
@@ -1032,33 +1053,94 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
+  - `.reuseaddr`: turn on SO_REUSEADDR flag?
+  - `.reuseport`: turn on SO_REUSEPORT flag?
+  - `.socket`: A cqueues socket object
+  - `.tls`: `nil`: allow both tls and non-tls connections
+  -         `true`: allows tls connections only
+  -         `false`: allows non-tls connections only
+  - `.version`: the http version to allow to connect (default: any)
+
+
+#### Example {#http.server.listen-example}
 
-### `listen(options)` <!-- --> {#http.server.connect}
+```lua
+
+local app_server = http_server.listen {
+		host = 192.168.0.1;
+		port = 80;
+		onstream = function(server, stream)  
+			local request_headers = assert(stream:get_headers())
+			local request_method = request_headers:get ":method"
+
+			print(string.format('[%s] "%s %s HTTP/%g"  "%s" "%s" ',
+				os.date("%d/%b/%Y:%H:%M:%S %z"),
+				request_headers:get(":method") or "",
+				request_headers:get(":path") or "",
+				stream.connection.version,
+				request_headers:get("referer") or "-",
+				request_headers:get("user-agent") or "-"
+				))
+		end;
+}
+
+```
 
 
 ### `server:onerror(new_handler)` <!-- --> {#http.server:onerror}
 
+Replaces the current error handler function with `new_handler` and returns a reference to the old function. `new_handler` is required to have at least one parameter. The default error handler is `nil`, and can be set during instantiation through the `options` table passed to the [*server.listen(options)*](#server.listen) function.
+
 
 ### `server:listen(timeout)` <!-- --> {#http.server:listen}
 
+Used to set up the local socket and initiate listening on the specified port. This function only needs to be called if you are not calling [*localname*](#http.server:localname) before [*step*](#http.server:step) or [*loop*](#http.server:loop). On error, returns nil, an error message and an error number.
+
 
 ### `server:localname()` <!-- --> {#http.server:localname}
 
+Returns the connection information for the local socket. Returns address family, IP address and port for an external socket. For Unix domain sockets, the function returns AF_UNIX and the path. If the connection object is not connected, returns AF_UNSPEC (0). On error, returns nil an error message and an error number.
+
 
 ### `server:pause()` <!-- --> {#http.server:pause}
 
-Cause the server loop to stop processing new clients until [`:resume`](#http.server:resume) is called.
+Cause the server loop to stop processing new clients until [`:resume`](#http.server:resume) is called. Existing client requests will run to completion. 
 
 
 ### `server:resume()` <!-- --> {#http.server:resume}
 
+Resumes a paused `server` and processes new client requests. 
+
 
 ### `server:close()` <!-- --> {#http.server:close}
 
+Shutdown the server and close the socket. A closed server cannot be reused.
+
 
 ### `server:pollfd()` <!-- --> {#http.server:pollfd}
 
+Resume the `server` from a yielded state and return a file descriptor integer or a variable object. Should return `nil` at a minimum on error.
```

Comment:
`:pollfd()` doesn't resume. It returns a file descriptor (as an integer) that can be passed to a system api like `select` or `kqueue` to wait on anything *this* server object wants to do. It's used for integrating with other main loops, and should be used in combination with `:events()` and `:timeout()`